code/datums/cleaning.dm

```diff
@@ -0,0 +1,52 @@
+//For handling standard click-to-clean items like soap and mops.
+
+//text1, text2, and text3 carry strings that, when pieced together by this proc, make the cleaning messages.
+//text1 and text2 pertain to the start message, i.e. "User begins to [clean] the floor[ with the mop.]"
+//text3 pertains to the end message, i.e. "You [finished mopping] the floor."
+
+/atom/proc/cleaning_act(mob/user, atom/cleaner, cleanspeed = 50, text1 = "clean", text2 = " with [cleaner].", text3 = "clean")
+	var/cmag_cleantime = 50 //The cleaning time for cmagged objects is locked to this, for balance reasons
+
+	if(user.client && (src in user.client.screen)) //You can't clean items you're wearing for technical reasons
+		to_chat(user, "<span class='notice'>You need to take that [name] off before cleaning it.</span>")
+		return FALSE
+
+	if(HAS_TRAIT(src, TRAIT_CMAGGED)) //So we don't need a cleaning_act for every cmaggable object
+		text1 = "clean the ooze off"
+		text3 = "clean the ooze off"
```

Comment:
could you use a locally scoped var such as `is_cmagged` for logic statements instead of `if(text3 == "clean the ooze off") /`